lib/t/feeds.ex

```diff
@@ -625,4 +629,149 @@ defmodule T.Feeds do
+
+  @doc """
+  Starts a "Since Live" event for newbies:
+
+    - sends a message to the admin tg bot
+    - broadcasts a push notification to all participants
+    - broadcasts a `:start` event to all newbies' feed channels
+
+  This function shouldn't be called directly, but rather scheduled with Oban in crontab.
+  """
+  @spec newbies_start_live(DateTime.t()) :: :ok
+  def newbies_start_live(reference \\ DateTime.utc_now()) do
+    newbies = newbies_list_today_participants(reference)
+
+    m = "LIVE (newbie's special edition) started: newbies count #{length(newbies)}"
+    Logger.warn(m)
+    Bot.async_post_message(m)
+
+    Enum.each(newbies, fn user_id ->
+      feed_channel_topic = "feed:" <> user_id
+      broadcast(feed_channel_topic, {__MODULE__, [:mode_change, :start]})
+    end)
+
+    newbies
+    |> T.Accounts.list_apns_devices()
+    |> DispatchJob.schedule_apns("newbie_live_mode_started", _data = %{})
+
+    :ok
+  end
+
+  @doc """
+  Ends a "Since Live" event for newbies:
+
+    - sends a message to the admin tg bot
+    - broadcasts a `:end` event to all `:mode_change` subscribers
+    - clears live tables
+
+  This function shouldn't be called directly, but rather scheduled with Oban in crontab.
+  """
+  @spec newbies_end_live :: :ok
+  def newbies_end_live do
+    m = "LIVE (newbie's special edition) ended"
+    Logger.warn(m)
+    Bot.async_post_message(m)
+
+    # we don't need to send `end` event only to newbies,
+    # we can turn off everyone who's live right now
+    # since "newbie live" and "normal live" don't intersect
+    notify_subscribers(:mode_change, :end)
+
+    two_hours_ago_ref = DateTime.add(DateTime.utc_now(), -7200)
+    clear_live_tables(two_hours_ago_ref)
+
+    :ok
+  end
+
+  @doc """
+  Lists ids of today's "newbies live" participants.
+
+  "newbies live" participants are:
+  - new newbies: users who have registered today (they are still newbies even if they've had a call)
+  - TODO old newbies: users who have registered before but haven't had a call
+  - oldies or "midwives": experienced users who explain newbies what all this is about
+  """
+  @spec newbies_list_today_participants :: [Ecto.UUID.t()]
+  def newbies_list_today_participants(reference \\ DateTime.utc_now()) do
+    # maybe Repo.stream later
+    # newbies = new_newbies_q(reference) |> union(^old_newbies_q()) |> Repo.all()
+    newbies = new_newbies_q(reference) |> Repo.all()
+    hardcoded_oldies() ++ newbies
+  end
+
+  defp hardcoded_oldies do
+    Application.get_env(:t, :oldies) || []
+  end
+
+  # TODO move to accounts.ex? maybe later
+  @spec new_newbies_q(DateTime.t()) :: Ecto.Query.t()
+  defp new_newbies_q(reference) do
+    T.Accounts.User
+    |> where([u], is_nil(u.blocked_at))
+    |> where([u], not is_nil(u.onboarded_at))
+    |> where([u], fragment("? + interval '24h' >= ?::timestamp ", u.onboarded_at, ^reference))
+    |> join(:inner, [u], p in FeedProfile, on: u.id == p.user_id and not p.hidden?)
```

Comment:
this fixes the issue with inviting users who didn't complete their stories, but now these users will never be invited to newbies LIVE
this maybe not so critical, but maybe fixed by introducing new column to users table (~~"real" onboarded.at~~, "onboarded_with_profile_at")?

Reply:
I've opened a separate PR for this: https://github.com/getsince/test3/pull/441